app/views/online-class/scheduler.vue

```diff
@@ -0,0 +1,148 @@
+<template>
+  <page-template>
+    <div class="scheduler">
+      <div class="header">
+        {{ header }}
+      </div>
+      <div class="body">
+        <class-info
+          v-if="step === 'class'"
+          :code-language-map="codeLanguageMap"
+          :levels="levels"
+          @change-class-info="changClassInfo"
+        />
+        <available-time
+          v-else-if="step === 'available'"
+          :class-info="classInfo"
+          @back="step = 'class'"
+          @next="selectTime"
+        />
+        <student-info
+          v-else-if="step === 'student'"
+          :class-details="classDetails"
+          @change-student-info="changeStudentInfo"
+          @back="step = 'available'"
+        />
+        <next-step
+          v-else
+        />
+      </div>
+    </div>
+  </page-template>
+</template>
+
+<script>
+import moment from 'moment'
+import pageTemplate from '../parents/PageTemplate.vue'
+import classInfo from './components/classInfo.vue'
+import availableTime from './components/availableTime.vue'
+import studentInfo from './components/studentInfo.vue'
+import nextStep from './components/nextStep.vue'
+
+import { tempBookTime, bookTime } from '../../core/api/online-classes'
+export default {
+  name: 'SchedulerView',
+  components: {
+    pageTemplate,
+    classInfo,
+    studentInfo,
+    nextStep,
+    availableTime
+  },
+  data () {
+    return {
+      header: 'Book Live Online Classes',
+      step: 'class',
+      classInfo: {},
+      time: {},
+      studentInfo: {},
+      codeLanguageMap: {
+        python: 'Python',
+        javascript: 'JavaScript',
+        html: 'HTML',
+        css: 'CSS',
+        lua: 'Lua',
+        java: 'Java',
+        cpp: 'C++',
+        coffeescript: 'CoffeeScript',
+      },
+      levels: ['Beginner', 'Intermediate', 'Advanced']
+    }
+  },
+  computed: {
+    classDetails () {
+      const lang = this.classInfo.language
+      const codeLang = this.codeLanguageMap[this.classInfo.codeLanguage]
+      const level = this.levels[this.classInfo.level]
+      const date = moment(this.time.date).set({ hour: this.time.time, minute: 0 })
```

Comment:
why are we assuming minute 0? It will be based on timezone, correct? Is it a TODO still?
Also, why not use date-time that is actually booked in db and returned from server to convert and show here?

Reply:
emmm let make a todo. both questions are related to timezone